src/lib.rs

```diff
@@ -26,25 +26,25 @@ mod url;
 
 type HandlerStream = Pin<Box<dyn Stream<Item = Result<Event, BroadcastStreamRecvError>> + Send>>;
 
-pub async fn run(cfg: Config, state: State) -> Result<(), Error> {
+pub async fn run(cfg: Config, state: State<'_>) -> Result<(), Error> {
     let tm_client = tendermint_rpc::HttpClient::new(cfg.tm_url.as_str()).map_err(Error::new)?;
     let mut app = App::new(tm_client, state);
 
     app = app.configure(cfg).await?;
     app.run().await
 }
 
-pub struct App {
+pub struct App<'a> {
     event_sub_client: event_sub::EventSubClient<tendermint_rpc::HttpClient>,
     event_sub_driver: event_sub::EventSubClientDriver,
     event_processor: EventProcessor,
     event_processor_driver: event_processor::EventProcessorDriver,
     state_updater: state::Updater,
-    state: State,
+    state: State<'a>,
 }
 
-impl App {
-    pub fn new(tm_client: tendermint_rpc::HttpClient, state: State) -> Self {
+impl<'a> App<'a> {
```

Comment:
```suggestion
impl App<'_> {
```
I think this should work, then you don't have to specify the lifetime in the fns anymore